File: PizzaOrderingSystem.Models/Classes/Orders/Order.cs
using PizzaOrderingSystem.Models.Classes.PurchaseCart;
using PizzaOrderingSystem.Models.Classes.Users;
using System;
using System.Collections.Generic;

namespace PizzaOrderingSystem.Models.Classes.Orders
{
    public enum OrderStatus
    {
        Ordered,
        Cooking,
        ReadyToPickUp,
        PickedUp,
        InTransit,
        Delivered,
        Cancelled,
        PaymentPending,
        PaymentDone,
        RefundInitiated,
        RefundDone
    }

    public class Order : BaseModel
    {
        public BaseModel User { get; set; }
        public List<OrderItem> OrderItems { get; set; }
        public Address DeliveryAddress { get; set; }
        public Address BillingAddress { get; set; }
        public DateTime OrderDate { get; set; }
        public OrderStatus Status { get; set; }
        public double OrderTotal { get; set; }
        public double Discount { get; set; }

        public Order()
        {
            OrderItems = new List<OrderItem>();
            Status = OrderStatus.Ordered;
            OrderTotal = 0;
            Discount = 0;
            OrderDate = DateTime.Now;

        }
        public Order(Cart cart)
        {
            OrderItems = new List<OrderItem>();
            OrderTotal = 0;
            Discount = 0;
            OrderDate = DateTime.Now;
            Status = OrderStatus.Ordered;
            foreach (var cartItem in cart.Items)
            {
                OrderItems.Add(new OrderItem(cartItem));
            }



        }
    }
}

File: PizzaOrderingSystem.SPA/src/app/models/Order/order.ts
import { BaseModel } from "../base-model";
import { Cart } from "../Cart/cart";
import { OrderItem } from "./order-item";

export class Order extends BaseModel {
    
    public items:OrderItem[];
    public price: number;
    constructor()
    {
        super()
        this.price = 0;
        this.items = new Array();
    }
    static CreateFrom(cart:Cart) {
        var result = new Order();
        result.price = cart.price;
        result.items = new Array();
        cart.items.forEach(cartItem => {
            result.items.push(OrderItem.CreateFrom(cartItem));
        });
        return result;
        }
}
